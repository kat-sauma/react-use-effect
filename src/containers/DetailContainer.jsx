import React, { Component } from 'react';
import DragDetail from '../components/app/dragRace/DragDetail';
import { fetchQueenById } from '../services/DragService';

export default class DetailContainer extends Component {
    state = {
      queen: { }
    }

    async componentDidMount() {
      const { id } = this.props.match.params;

      const queen = await fetchQueenById(id);

      this.setState({
        queen
      });
    }
    render() {
      console.log(this.state.queen, 'hi');
      const { queen } = this.state;
      return <DragDetail queen={queen} />;
    }
}
