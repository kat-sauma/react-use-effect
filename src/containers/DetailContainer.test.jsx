import React from 'react';
import { render, screen } from '@testing-library/react';
import DetailContainer from './DetailContainer';
import { MemoryRouter } from 'react-router-dom';

describe('DetailContainer', () => {
  it('renders details of queens to the page', async () => {
    render(<MemoryRouter><DetailContainer match={{ 
      params: { 
        id: 1
      }
    }}/></MemoryRouter>);

    const spinner = await screen.findByRole('spinner', { name: 'loading spinner' });

    expect(spinner).toMatchSnapshot();
  });
});
