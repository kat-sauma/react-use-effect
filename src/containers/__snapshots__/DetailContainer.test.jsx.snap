// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailContainer renders details of queens to the page 1`] = `
<img
  aria-label="loading spinner"
  role="spinner"
  src="public/loadingedit3.gif"
/>
`;

exports[`DetailContainer renders details of queens to the page 2`] = `
<section
  aria-label="queen info"
  class="queen-detail"
  role="listitem"
>
  <a
    href="/"
  >
    <img />
    <p />
    <p />
    <p />
    <p />
  </a>
</section>
`;
